### ISSUE TITLE
Using `live_redirect/2` in `Phoenix.LiveView` results in `handle_params/3` being called twice

### ISSUE THREAD

**Issue by tgxworld** [phoenixframework/phoenix_live_view#268]
### Environment

* Elixir version (elixir -v): Elixir 1.9.0 (compiled with Erlang/OTP 20)
* Phoenix version (mix deps): 1.4.8
* NodeJS version (node -v): v11.12.0
* NPM version (npm -v): 6.7.0
* Operating system: Ubuntun 18.04

### Actual behavior

I was trying out `live_redirect/2` when I noticed that using it in `handle_event` results in `handle_params/3` being called twice each time an event is received. I was able to reproduce the problem in https://github.com/chrismccord/phoenix_live_view_example with the `/users` live route. When using the `left` and `right` keydown events, the following stacktrace can be seen from within `handle_params`.

```
{:current_stacktrace,
 [
   {Process, :info, 2, [file: 'lib/process.ex', line: 767]},
   {DemoWeb.UserLive.Index, :handle_params, 3,
    [file: 'lib/demo_web/live/user_live/index.ex', line: 20]},
   {Phoenix.LiveView.Channel, :sync_handle_params_with_live_redirect, 4,
    [file: 'lib/phoenix_live_view/channel.ex', line: 301]},
   {Phoenix.LiveView.Channel, :handle_result, 3,
    [file: 'lib/phoenix_live_view/channel.ex', line: 202]},
   {:gen_server, :try_dispatch, 4, [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}
 ]}
[debug] QUERY OK source="users" db=0.8ms
SELECT u0."id", u0."username", u0."email", u0."phone_number", u0."inserted_at", u0."updated_at" FROM "users" AS u0 ORDER BY u0."id" LIMIT $1 OFFSET $2 [5, 5]
{:current_stacktrace,
 [
   {Process, :info, 2, [file: 'lib/process.ex', line: 767]},
   {DemoWeb.UserLive.Index, :handle_params, 3,
    [file: 'lib/demo_web/live/user_live/index.ex', line: 20]},
   {Phoenix.LiveView.Channel, :handle_info, 2,
    [file: 'lib/phoenix_live_view/channel.ex', line: 68]},
   {:gen_server, :try_dispatch, 4, [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}
 ]}
[debug] QUERY OK source="users" db=0.2ms
SELECT u0."id", u0."username", u0."email", u0."phone_number", u0."inserted_at", u0."updated_at" FROM "users" AS u0 ORDER BY u0."id" LIMIT $1 OFFSET $2 [5, 5]
{:current_stacktrace,
 [
   {Process, :info, 2, [file: 'lib/process.ex', line: 767]},
   {DemoWeb.UserLive.Index, :handle_params, 3,
    [file: 'lib/demo_web/live/user_live/index.ex', line: 20]},
   {Phoenix.LiveView.Channel, :sync_handle_params_with_live_redirect, 4,
    [file: 'lib/phoenix_live_view/channel.ex', line: 301]},
   {Phoenix.LiveView.Channel, :handle_result, 3,
    [file: 'lib/phoenix_live_view/channel.ex', line: 202]},
   {:gen_server, :try_dispatch, 4, [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}
 ]}
[debug] QUERY OK source="users" db=2.0ms
SELECT u0."id", u0."username", u0."email", u0."phone_number", u0."inserted_at", u0."updated_at" FROM "users" AS u0 ORDER BY u0."id" LIMIT $1 OFFSET $2 [5, 10]
{:current_stacktrace,
 [
   {Process, :info, 2, [file: 'lib/process.ex', line: 767]},
   {DemoWeb.UserLive.Index, :handle_params, 3,
    [file: 'lib/demo_web/live/user_live/index.ex', line: 20]},
   {Phoenix.LiveView.Channel, :handle_info, 2,
    [file: 'lib/phoenix_live_view/channel.ex', line: 68]},
   {:gen_server, :try_dispatch, 4, [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 249]}
 ]}
[debug] QUERY OK source="users" db=2.0ms
SELECT u0."id", u0."username", u0."email", u0."phone_number", u0."inserted_at", u0."updated_at" FROM "users" AS u0 ORDER BY u0."id" LIMIT $1 OFFSET $2 [5, 10]
```

### Expected behavior

`handle_params/3` called once on `live_redirect`.

**Comment by JoeZ99** [comment]
Same here.

  - Elixir 1.9.0 (compiled with Erlang/OTP 22)
  - Phoenix 1.4.7

**Comment by chrismccord** [comment]
Fixed via #248

**Comment by tgxworld** [comment]
Thank you @chrismccord :+1:
